### PR TITLE
Reject out-of-range SHE counter, flags and key IDs

### PR DESCRIPTION
**Pull request by yosuke-wolfssl**
### Problem
The SHE key-update builders expose packed protocol fields as full-width C
integers and validate only pointers and buffer lengths, never value ranges.
Out-of-range values were silently truncated or bled into the adjacent field,
and the call still returned 0:

| Parameter | C type | Field width | Effect when exceeded |
|---|---|---|---|
| `counter` | `word32` | 28 bits | wraps to a smaller value |
| `flags` | `byte` | 4 bits | overwrites the counter's low nibble |
| `authKeyId` | `byte` | 4 bits | bleeds into the target-key-ID nibble |
| `targetKeyId` | `byte` | 4 bits | truncated |

The counter is SHE's anti-rollback field, and `wc_SHE_GetCounter()` hands the
caller a `word32` that is fed straight back in, so the wrap is reachable on a
normal round trip. A key ID above `0x0F` is worse: the generated M1 or M4 names
a different HSM key slot than the caller asked for, with no error returned.
Medium severity.

### Fix (`wolfcrypt/src/wc_she.c`)
`wc_SHE_GenerateM1M2M3()` and `wc_SHE_GenerateM4M5()` now reject these with
`BAD_FUNC_ARG`:

```c
    /* Reject values wider than their packed field */
    if (counter > WC_SHE_COUNTER_MAX || flags > WC_SHE_FLAGS_MAX ||
        authKeyId > WC_SHE_KEY_ID_MAX || targetKeyId > WC_SHE_KEY_ID_MAX) {
        return BAD_FUNC_ARG;
    }
```

- **Placement is at function entry, ahead of the crypto-callback dispatch**,
  unlike the existing pointer/size checks. The widths come from the SHE
  protocol, so an HSM backend handed a 2^28 counter would build an equally
  wrong message. There is no "the secure element supplies it instead" case for
  a value range the way there is for `authKey`/`newKey` being `NULL`.
- **Rejecting, not masking.** Masking would still change the caller's intent
  without saying so.
- **`WC_SHE_COUNTER_MAX`, `WC_SHE_FLAGS_MAX`, `WC_SHE_KEY_ID_MAX`** added to
  `wc_she.h` beside the existing shift constants.
- Checks the encodable 4-bit width, not the narrower slot policy the header
  prose mentions (auth 0-14, target 1-14), since slot numbering is
  HSM-specific.

In-tree callers are unaffected: `wc_SHE_LoadKey_Internal()` passes zeros and
the KATs pass small values.

Closes f-7077.

### Tests (`tests/api/test_she.c`)
Seven negative cases in `test_wc_SHE_DecisionCoverage()`, one out-of-range
operand per call, plus a boundary-accept case in
`test_wc_SHE_GenerateM1M2M3()` proving the largest encodable values still
succeed and the comparison is `>` and not `>=`.

### Verification
- Negative control: before the fix, `test_wc_SHE_DecisionCoverage` failed with
  `result: 0 != -173`. After, the `she` group is 14/14 passed, 0 failed.
- Full `./tests/unit.test` and `./wolfcrypt/test/testwolfcrypt` pass.
- Builds clean with no new warnings under both `--enable-she=standard` (no
  `WOLF_CRYPTO_CB`) and `--enable-she=extended --enable-cryptocb`.

### CODE REVIEW

File: wolfcrypt/src/wc_she.c [hunk 1]
```diff
@@ -501,6 +501,12 @@ int wc_SHE_GenerateM1M2M3(wc_SHE* she,
         return BAD_FUNC_ARG;
     }
 
+    /* Reject values wider than their packed field */
+    if (counter > WC_SHE_COUNTER_MAX || flags > WC_SHE_FLAGS_MAX ||
```
[the remark this answers is not in the document]
Comment (yosuke-wolfssl) [reply wolfcrypt/src/wc_she.c:505]: Accurate for the software path, but keeping the check unconditional here.

The range check sits *above* the crypto-callback dispatch. `m2pOverride`/`m4pOverride` only affect what `she_build_headers()` writes on the software path — the callback receives `counter` and `flags` verbatim regardless of either override. Skipping validation when an override is set would let an out-of-range counter or flags value through to an HSM callback, which is exactly the silent truncation this PR closes.

Separately, a counter wider than 28 bits or a key ID wider than 4 is a caller bug whether or not this particular call happens to consume the value, so rejecting it is the contract we want rather than an inconsistency to smooth over. Leaving as-is.

File: tests/api/test_she.c [hunk 2]
```diff
@@ -862,6 +890,19 @@ int test_wc_SHE_DecisionCoverage(void)
         ExpectIntEQ(wc_SHE_GenerateM4M5(&she, uid, WC_SHE_UID_SZ, 1, 2, key,
             WC_SHE_KEY_SZ, 1, m4, sizeof(m4), m5, WC_SHE_M5_SZ - 1),
             WC_NO_ERR_TRACE(BAD_FUNC_ARG));
+
+        /* wc_SHE_GenerateM4M5: packed field widths, one invalid per call. */
+        ExpectIntEQ(wc_SHE_GenerateM4M5(&she, uid, WC_SHE_UID_SZ, 1, 2, key,
```
[the remark this answers is not in the document]
Comment (yosuke-wolfssl) [reply tests/api/test_she.c:903]: Fixed. Added an accept-at-maximum case to `test_wc_SHE_GenerateM4M5`: `WC_SHE_COUNTER_MAX` with `WC_SHE_KEY_ID_MAX` for both key IDs, asserting a 0 return and `m4[WC_SHE_M4_KID_OFFSET] == 0xFF` so the packed ID byte is checked too, mirroring the existing M1M2M3 boundary case.

Confirmed it is a real gate: flipping the `wc_SHE_GenerateM4M5` guard from `>` to `>=` fails `test_wc_SHE_GenerateM4M5` and nothing else in the suite. Clean at 14/14 with the guard restored.

[the remark this answers is not in the document]
Comment (yosuke-wolfssl) [reply tests/api/test_she.c:903]: Duplicate of the finding directly above; covered by the same M4M5 boundary case.

**Review by philljj — APPROVED** [review]
